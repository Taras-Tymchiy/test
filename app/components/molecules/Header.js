import React from 'react';
import {
  Platform,
  StyleSheet,
  View
} from 'react-native';
import SimpleButton from '../atoms/SimpleButton';
import { TextHeader } from '../atoms/Text';
import styles from './Header.styles';

const APPBAR_HEIGHT = Platform.OS === 'ios' ? 44 : 56;
const STATUSBAR_HEIGHT = Platform.OS === 'ios' ? 20 : 0;
const TITLE_OFFSET = Platform.OS === 'ios' ? 70 : 56;

type HeaderProps = {
  title?: string;
  leftButtonTitle?: string;
  rightButtonTitle?: string;
  onLeftButtonPress?: ()=> void;
  onRightButtonPress?: ()=> void,
}

class Header extends React.PureComponent<HeaderProps> {
  render() {
    const containerStyles = [ styles.container, { height: APPBAR_HEIGHT } ];
    const { 
      title,
      leftButtonTitle, 
      rightButtonTitle, 
      onLeftButtonPress,
      onRightButtonPress
    } = this.props;
    return (
      <View
        style={containerStyles}
        forceInset={{ top: 'always', bottom: 'never' }}
      >
        <View style={[StyleSheet.absoluteFill, styles.header]} >
          <SimpleButton style={styles.leftButton} title={leftButtonTitle} onPress={onLeftButtonPress} />
          <View>
            <TextHeader>{title}</TextHeader>
          </View>
          <SimpleButton style={styles.rightButton} textStyle={styles.rightButtonText} title={rightButtonTitle} onPress={onRightButtonPress} />
        </View>
      </View>
    );
  }
}

export default Header;