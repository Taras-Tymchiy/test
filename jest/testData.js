import { Post } from "../app/entities/Post";


export const posts: Post[] = [
    {
        id: 1,
        id_str: '1',
        created_at: '2015-01-01',
        user: {
            id: 1,
            name: 'John Snow',
        },
        text: 'post text 1'
    },
    {
        id: 2,
        id_str: '2',
        created_at: '2015-01-02',
        user: {
            id: 2,
            name: 'Barack Obama',
        },
        text: 'post text 2'
    }
]