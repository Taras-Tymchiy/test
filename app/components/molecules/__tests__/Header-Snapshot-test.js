import React from 'react';
import * as jest from 'jest';
import renderer from 'react-test-renderer';
import Header from '../Header';
import { View } from 'react-native';

const props: HeaderProps = {
    title: 'Test title',
    leftButtonTitle: 'Test left title',
    rightButtonTitle: 'Test right title',
    onLeftButtonPress: ()=> null,
    onRightButtonPress: ()=> null,
};

it('renders a Header using Snapshots', () => {
  expect(renderer.create(
    <Header {...props} />
  )).toMatchSnapshot();
});
