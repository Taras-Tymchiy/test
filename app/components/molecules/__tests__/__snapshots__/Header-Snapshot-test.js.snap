// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Header using Snapshots 1`] = `
<View
  forceInset={
    Object {
      "bottom": "never",
      "top": "always",
    }
  }
  style={
    Array [
      Object {
        "backgroundColor": "#F7F7F7",
        "borderBottomColor": "rgba(0, 0, 0, .3)",
        "borderBottomWidth": 0.5,
      },
      Object {
        "height": 64,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 20,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingLeft": 12,
          "width": 70,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#037aff",
              "fontSize": 16,
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        Test left title
      </Text>
    </View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#222",
              "fontSize": 20,
              "fontWeight": "bold",
            },
            undefined,
          ]
        }
      >
        Test title
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingRight": 12,
          "width": 70,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#037aff",
              "fontSize": 16,
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "bold",
              "textAlign": "right",
            },
          ]
        }
      >
        Test right title
      </Text>
    </View>
  </View>
</View>
`;
